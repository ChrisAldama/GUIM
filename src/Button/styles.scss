@import '../styles/themes.scss';

.GUIMButton {
  border-radius: 5px;
  padding: 10px 15px;
  cursor: pointer;
  transition: all ease 0.3s;
  font-size: 0.8em;
  font-weight: 300;
  user-select: none;
  text-align: center;
  text-decoration: none;
  &:hover, &:focus{
    opacity: 0.75;
  };
}

.GUIMButtonBlue {
  @apply --blue-theme;
}

.GUIMButtonGray {
  @apply --gray-theme;
}

.GUIMButtonGreen {
  @apply --green-theme;
}
