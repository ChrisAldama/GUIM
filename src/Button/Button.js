import React from 'react';
import './styles.scss';

const Button = props => (
  <a
    onClick={props.onClick}
    className={props.className}>
    {props.label}
  </a>
);

Button.defaultProps = {
  label: 'Click me',
  onClick: () => console.log('Click on button'),
  className: 'GUIMButton'
};

export default Button;
