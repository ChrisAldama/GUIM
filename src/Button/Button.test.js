import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Button />', () => {
  let clicked = false;
  let tree;
  let reRender;
  const makeRenderer = component =>
    () => tree = component.toJSON();


  beforeEach(() => {
    component = renderer.create(
    <Button
      color='gray'
      onClick={() => clicked = true}
    />)
    reRender = makeRenderer(component);
    reRender();
  });

  it('match snapshot', () => expect(tree).toMatchSnapshot());

  it('is true after click', () => {
    tree.props.onClick();
    reRender();
    expect(clicked).toBe(true);
  });

});
