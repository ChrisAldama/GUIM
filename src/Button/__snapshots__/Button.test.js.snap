// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> match snapshot 1`] = `
<a
  className="GUIMButton  GUIMButtonGRAY"
  onClick={[Function]}
>
  Click me
</a>
`;
